screen/wallets/entropyGenerator.js

```diff
@@ -0,0 +1,426 @@
+    newPaths[paths.length - 1].segments.push(`${mode} ${g.x} ${g.y}`);
+    setPaths(newPaths);
+  };
+
+  const pan = Gesture.Pan()
+    .onStart((g) => {
+      onDrawing('M', g);
+    })
+    .onUpdate((g) => {
+      onDrawing('L', g);
+    })
+    .onEnd(() => {
+      setPaths([...paths, { segments: [], color }]);
+    })
+    .minDistance(1);
+
+  const onClearDrawingButtonClick = () => {
+    setPaths([{ segments: [], color }]);
+  };
+
+  const { colors } = useTheme();
+  const { addWallet, saveToDisk, isAdancedModeEnabled, wallets, isTestModeEnabled } = useContext(BlueStorageContext);
+  const [isLoading, setIsLoading] = useState(true);
+  const [walletBaseURI, setWalletBaseURI] = useState('');
+  const [selectedIndex, setSelectedIndex] = useState(0);
+  const [label, setLabel] = useState(labelProps || '');
+  const [isAdvancedOptionsEnabled, setIsAdvancedOptionsEnabled] = useState(false);
+  const [isTestMode, setIsTestMode] = useState(false);
+  const [selectedWalletType, setSelectedWalletType] = useState(selectedWalletTypeProps || false);
+  const [backdoorPressed, setBackdoorPressed] = useState(1);
+
+  const [entropyButtonText, setEntropyButtonText] = useState(loc.wallets.add_entropy_provide);
+  const stylesHook = {
+    advancedText: {
+      color: colors.feeText,
+    },
+    label: {
+      borderColor: colors.formBorder,
+      borderBottomColor: colors.formBorder,
+      backgroundColor: colors.inputBackgroundColor,
+    },
+    noPadding: {
+      backgroundColor: colors.elevated,
+    },
+    root: {
+      backgroundColor: colors.elevated,
+      marginBottom: 30,
+    },
+    lndUri: {
+      borderColor: colors.formBorder,
+      borderBottomColor: colors.formBorder,
+      backgroundColor: colors.inputBackgroundColor,
+    },
+  };
+
+  useEffect(() => {
+    AsyncStorage.getItem(isTestMode ? AppStorage.TEST_LNDHUB : AppStorage.LNDHUB)
+      .then((url) => setWalletBaseURI(url || (isTestMode ? 'https://tnlndhub.mintlayer.org' : 'https://lndhub.io')))
+      .catch(() => setWalletBaseURI(''));
+    isTestModeEnabled().then(setIsTestMode);
+    isAdancedModeEnabled()
+      .then(setIsAdvancedOptionsEnabled)
+      .finally(() => setIsLoading(false));
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [isAdvancedOptionsEnabled]);
+
+  const generateEntropyDrawing = async () => {
+    const flatArray = paths.flatMap((path) => path.segments);
+    const numArr = flatArray.flatMap((str) => {
+      // Split the string into parts by spaces
+      const parts = str.split(' ');
+      // Convert the numerical parts into numbers
+      const nums = parts.slice(1).map(parseFloat);
+      // Return an array with the first part and the numbers
+      return nums;
+    });
+    const normalizedPoints = normalize(numArr.map((point) => Math.round(point)));
+    return generateEntropy(normalizedPoints);
```

Comment:
can we add a comment somewhere that explains the lib will xor the user "randomness" with a prng so we don't bother this side